### ISSUE TITLE
UNAVAILABLE: Internal: output 'labels' does not follow naming convention i.e. <name>__<index>.

### ISSUE THREAD

**Issue by zhaozhiming37** [triton-inference-server/server#4557]
**Description**
My torchscript model output is a Dict[str, Tensor], and my config is as follows

```
platform: "pytorch_libtorch"
max_batch_size : 16
input [
  {
    name: "input__0"
    data_type: TYPE_FP32
    format: FORMAT_NCHW
    dims: [ -1, -1, 3 ]
  }
]
output [
  {
    name: "labels"
    data_type: TYPE_INT64
    dims: [ -1, 1 ]
  }
]
output [
  {
    name: "scores"
    data_type: TYPE_FP32
    dims: [ -1, 1 ]
  }
]
output [
  {
    name: "boxes"
    data_type: TYPE_FP32
    dims: [ -1, 4 ]
  }
]
model_warmup [
  {
    name: "zero_tensor"
    batch_size: 1
    inputs {
      key: "input__0"
      value: {
        data_type: TYPE_FP32
        dims: [1080, 1920, 3]
        zero_data: true
      }
    }
  },
  {
    name: "random_tensor"
    batch_size: 1
    inputs {
      key: "input__0"
      value: {
        data_type: TYPE_FP32
        dims: [1080, 1920, 3]
        random_data: true
      }
    }
  }
]
```

My issue is:
`UNAVAILABLE: Internal: output 'labels' does not follow naming convention i.e. <name>__<index>. `.
However `'labels'` is in my model output dict.I don't know what's wrong.

**Triton Information**
My docker env: `nvcr.io/nvidia/tritonserver:22.03-py3`

**To Reproduce**
- Model backend is pytorch
- Input is a Tensor with shape [N, -1, -1, 3]
- Output is a dict :`{'labels':Tensor[N, -1, 1], '':Tensor[N, -1, 1], '':Tensor[N, -1, 4]}`  

1. **save torchscript model**

```
import torch
import torch.nn as nn


class MyModel(nn.Module):
    def __init__(self):
        super(MyModel, self).__init__()

    def forward(self, x):
        return {'labels': x}


if __name__ == '__main__':
    script_path = './model.pt'
    device = torch.device('cuda:0')
    mymodel = MyModel()
    script = torch.jit.script(mymodel)
    script.eval()
    script = script.to(device)
    script.save(script_path)
```

2. **deploy model by triton**

```
platform: "pytorch_libtorch"
max_batch_size : 16
input [
  {
    name: "input__0"
    data_type: TYPE_FP32
    format: FORMAT_NCHW
    dims: [ -1, -1, 3 ]
  }
]
output [
  {
    name: "labels"
    data_type: TYPE_INT64
    dims: [ -1, 1 ]
  }
]
model_warmup [
  {
    name: "zero_tensor"
    batch_size: 1
    inputs {
      key: "input__0"
      value: {
        data_type: TYPE_FP32
        dims: [1080, 1920, 3]
        zero_data: true
      }
    }
  },
  {
    name: "random_tensor"
    batch_size: 1
    inputs {
      key: "input__0"
      value: {
        data_type: TYPE_FP32
        dims: [1080, 1920, 3]
        random_data: true
      }
    }
  }
]
```

**Expected behavior**
When my model outputs a dictionary, does the output name in the configuration file match the dictionary key?

**Comment by rmccorm4** [comment]
@CoderHam can you chime in on this? 

This [section](https://github.com/triton-inference-server/server/blob/main/docs/model_configuration.md#special-conventions-for-pytorch-backend) seems to imply this config is OK for inputs, but I'm not sure if it mentions the same for outputs.

**Comment by CoderHam** [comment]
@rmccorm4 is right, the PyTorch backend does not support Dictionary as output. 
https://github.com/triton-inference-server/server/blob/r22.03/docs/model_configuration.md#special-conventions-for-pytorch-backend

@zhaozhiming37 The dictionary support is limited to inputs and you must modify your model to return the tensor itself instead of a dictionary.
